pkgs/stdenv/generic/setup.sh

```diff
@@ -685,39 +672,72 @@ substitute() {
                 ;;
 
             *)
-                echo "substitute(): ERROR: Invalid command line argument: $1" >&2
+                echo "substituteStream(): ERROR: Invalid command line argument: $1" >&2
                 return 1
                 ;;
         esac
 
-        content="${content//"$pattern"/$replacement}"
+        eval "$var"'=${'"$var"'//"$pattern"/"$replacement"}'
```

Comment:
What’s this commit do? Such major changes to the internals of `stdenv` should *at least* have an extensive commit message of all the implications. I’m told `substituteInPlace` is very slow (in the order of seconds), so on inquiring this is the last commit that popped up, with no commit description whatsoever. :(

Reply:
Was this resolved? Does this still have a performance problem? (If so, is there a known way to reproduce it?)

Reply:
To reproduce it, try changing something in `pkgs/development/tools/build-managers/bazel/default.nix` and then run `nix-build -A bazel /path/to/nixpkgs`.

You should see that the substitution of every file takes like half a second, which is a lot for a simple text substitution.

See also https://github.com/NixOS/nixpkgs/pull/35304#issuecomment-445550010, cc @mboes

Reply:
It's always been O(size of file * number of @var@ definitions), which is quiet shitty.

Just as we did for "expand response params", we should probably rewrite this in C or something.